src/components/Sheet/Sheet.scss

```diff
@@ -0,0 +1,71 @@
+$sheet-desktop-width: rem(380px);
+
+.Sheet {
+  position: fixed;
+  bottom: 0;
+  width: 100vw;
+  height: 100%;
+  background: color('white');
+  box-shadow: shadow('layer');
+
+  &.desktop {
+    right: 0;
+    width: $sheet-desktop-width;
+  }
+
+  &:focus {
+    outline: 0;
+  }
+}
+
+.Container {
+  position: fixed;
+  z-index: z-index('modal', $fixed-element-stacking-order);
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+
+  &.desktop {
+    left: auto;
+    width: $sheet-desktop-width;
+  }
+}
+
+// Bottom
+.Bottom {
+  will-change: transform;
+  transition: transform duration('slow') easing('base');
+  transform-origin: bottom;
+}
+.enterBottom {
+  transform: translateY(100%);
+}
+.enterBottomActive {
+  transform: translateY(0%);
+}
+.exitBottom {
+  transform: translateY(0%);
+}
+.exitBottomActive {
+  transform: translateY(100%);
+}
+
+// Right
+.Right {
+  will-change: transform;
+  transition: transform duration('slow') easing('base');
+  transform-origin: right;
+}
+.enterRight {
+  transform: translateX(100%);
+}
+.enterRightActive {
+  transform: translateX(0%);
+}
+.exitRight {
+  transform: translateX(0%);
+}
+.exitRightActive {
+  transform: translateX(100%);
+}
```

Comment:
I've optimized the transition code for `Bottom` and `Right` (used [here](https://github.com/Shopify/polaris-react/pull/1250/files#diff-a9e19997da37d37567719a0d15889f25R102)) for ease of extraction into some sort of `Motion` component. This is something I'm considering digging into for hackdays, but I'd like to keep that functionality localized here until that's done.